encodings/dict/benches/dict_compress.rs

```diff
@@ -1,71 +1,87 @@
-#![allow(clippy::unwrap_used)]
-#![allow(clippy::cast_possible_truncation)]
-
-use criterion::{criterion_group, criterion_main, Criterion, Throughput};
-use rand::prelude::StdRng;
-use rand::SeedableRng;
+use divan::Bencher;
+use rand::distributions::Standard;
+use rand::prelude::Distribution;
 use vortex_array::array::{VarBinArray, VarBinViewArray};
 use vortex_array::IntoCanonical;
 use vortex_dict::builders::dict_encode;
 use vortex_dict::test::{gen_primitive_for_dict, gen_varbin_words};
+use vortex_dtype::NativePType;
+
+fn main() {
+    divan::main();
+}
 
-fn encode(c: &mut Criterion) {
-    let mut group = c.benchmark_group("dict_encode");
+const BENCH_ARGS: &[(usize, usize)] = &[
+    // length, unique_values
+    (100_000, 32),
+    (100_000, 128),
+    (100_000, 512),
+    (100_000, 2048),
+    (100_000, 8192),
+    (100_000, 32768),
```

Comment:
I know we don't, but it might be worth justifying the values we use? I don't think we expect to see the size

Reply:
You mean the value count I assume? Happy to drop some cases here. 

>  I don't think we expect to see the size

Where does that expectation come from? As long as the benchmarks are snappy, we could consider even benchmarking extreme cases.

Reply:
We don't think we should have array with more than 1 maybe 2k values (not 32k)

Reply:
I would prefer to have 2, 4, 16 included than 32k

Reply:
Will shift the inputs here accordingly. Agreed that it'd be good to have some small values included.